unit-tests/py/rspy/devices.py

```diff
@@ -132,13 +133,69 @@ def query( monitor_changes = True ):
             else:
                 sn = dev.get_info( rs.camera_info.serial_number )
             device = Device( sn, dev )
+            if device.port is not None:
+                known_ports.append(device.port)
             _device_by_sn[sn] = device
             log.d( '... port {}:'.format( device.port is None and '?' or device.port ), dev )
     finally:
         log.debug_unindent()
     #
     if monitor_changes:
         _context.set_devices_changed_callback( _device_change_callback )
+    if acroname:
+        need_recover = False
+        for dev in _device_by_sn.values():
+            if dev.port is None:
+                need_recover = True
+        if need_recover:
+            log.d( 'trying do discover unknown ports' )
+            ports = acroname.ports()
+            unknown_ports = [port for port in ports if port not in known_ports]
+            print( "ports:", ports )
+            print( "known ports:", known_ports )
+            print( "unknown ports:", unknown_ports )
+            for known_port in known_ports:
+                if known_port not in ports:
+                    log.e( "ERROR MESSAGE" ) # log.f?
+            if len( unknown_ports ) == 1:
+                for device in _device_by_sn.values():
+                    if device.port is None:
+                        log.d( 'port', unknown_ports[0], 'has device', device.serial_number )
+                        device._port = unknown_ports[0]
+            else:
+                # disabling all ports
+                acroname.disable_ports( ports )
+                for retry in range( 5 ):
```

Comment:
This repeats -- I suggest you make a helper fn with a `timeout` argument

Reply:
`disable_all()`?

Reply:
the problem with adding a helper is that every time the condition to break is different. I could add a helper that specifically waits for `enabled()` to be of some size. This will solve this instance of repetition but not others.
Are you saying I should use or add disable_all()? because I can't find it

Reply:
I'm saying you should add a `disable_all()` function that does this for and the following if